### PR TITLE
IP SANs in Kubelet serving cert

### PR DESCRIPTION
**Pull request by UtheMan**
Adds IP SANs to Kubelet serving cert to allow components to use secure TLS.

### CODE REVIEW

File: parts/linux/cloud-init/artifacts/cse_config.sh [hunk 1]
```diff
@@ -81,9 +82,37 @@ configureHTTPProxyCA() {
...
+    nsCertType = server
+    nsComment = "OpenSSL Generated Server Certificate"
+    subjectKeyIdentifier = hash
+    authorityKeyIdentifier = keyid,issuer:always
+    keyUsage = critical, digitalSignature, keyEncipherment
+    extendedKeyUsage = serverAuth
+    subjectAltName = @alt_names
+    [alt_names]
+    DNS.1 = ${NODE_NAME}
+    DNS.2 = ${NODE_IP}
```
Comment (alexeldeib) [inline parts/linux/cloud-init/artifacts/cse_config.sh:108]: do you happen to have a node with these changes still? can you share the output from `cat $KUBELET_SERVER_CERT_PATH | openssl x509 -noout -text`?

Comment (alexeldeib) [reply parts/linux/cloud-init/artifacts/cse_config.sh:108]: pr looks good to me, just wanted to sanity check how the final cert looks.

cc @phealy

Comment (UtheMan) [reply parts/linux/cloud-init/artifacts/cse_config.sh:108]: Sure here it is:
# cat $KUBELET_SERVER_CERT_PATH | openssl x509 -noout -text
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            22:ba:a1:5f:5e:40:95:64:ee:e5:86:8f:88:fe:c7:6e:8d:cc:da:a5
        Signature Algorithm: sha256WithRSAEncryption
        Issuer: CN = aks-nodes-38243387-vmss000000
        Validity
            Not Before: Jan 21 23:38:51 2022 GMT
            Not After : Jan 16 23:38:51 2042 GMT
        Subject: CN = aks-nodes-38243387-vmss000000
        Subject Public Key Info:
            Public Key Algorithm: rsaEncryption
                RSA Public-Key: (2048 bit)
                Modulus:
                    00:b9:ad:af:9f:52:05:fb:6f:a7:bc:6e:cc:ec:0b:
                    86:bc:82:cc:6d:7f:32:67:c3:df:0a:1a:7f:eb:d7:
                    0a:2a:d7:b6:be:20:9f:6c:85:54:bb:82:16:59:d9:
                    30:76:21:3b:3d:b4:6b:98:d1:c0:07:f9:8d:e6:d2:
                    c9:bf:a3:00:90:e7:cc:94:55:49:27:49:14:7c:1d:
                    2c:22:4c:38:a6:92:25:48:82:ec:b4:fb:9c:a0:86:
                    db:0d:6d:cb:62:1d:76:9e:74:93:cb:e9:8c:9a:f1:
                    db:93:1a:d6:ff:b2:c6:3f:56:9c:e4:dc:95:b2:62:
                    9a:08:fa:72:33:ea:f2:ab:ec:fa:c6:a2:df:86:4c:
                    9b:e9:cd:ff:1f:93:2d:83:34:e9:7b:45:95:ff:5d:
                    66:9f:5c:e6:a7:95:f6:c1:53:68:7c:c2:03:61:f1:
                    cf:9a:27:71:09:18:c3:be:4b:b0:ce:ea:a4:4f:ed:
                    53:e1:af:36:81:8f:07:0c:c1:a3:9f:42:71:7c:2c:
                    bb:a5:24:bf:4f:7f:6b:98:46:6e:50:04:9a:f7:cc:
                    8f:b3:9a:f6:7e:8f:cb:0d:b4:b9:40:9b:69:9f:8d:
                    3c:e5:6e:69:bd:04:52:26:b0:32:a7:be:55:1a:a2:
                    af:1b:54:f8:a2:78:65:63:14:7d:b3:94:bb:32:65:
                    04:f1
                Exponent: 65537 (0x10001)
        X509v3 extensions:
            X509v3 Basic Constraints: 
                CA:FALSE
            Netscape Cert Type: 
                SSL Server
            Netscape Comment: 
                OpenSSL Generated Server Certificate
            X509v3 Subject Key Identifier: 
                03:5F:B3:A3:CB:3C:C7:5E:53:92:41:55:E3:71:D4:90:5D:1C:F0:51
            X509v3 Authority Key Identifier: 
                keyid:03:5F:B3:A3:CB:3C:C7:5E:53:92:41:55:E3:71:D4:90:5D:1C:F0:51
                DirName:/CN=aks-nodes-38243387-vmss000000
                serial:22:BA:A1:5F:5E:40:95:64:EE:E5:86:8F:88:FE:C7:6E:8D:CC:DA:A5

            X509v3 Key Usage: critical
                Digital Signature, Key Encipherment
            X509v3 Extended Key Usage: 
                TLS Web Server Authentication
            X509v3 Subject Alternative Name: 
                DNS:aks-nodes-38243387-vmss000000, DNS:10.240.0.4, IP Address:10.240.0.4
    Signature Algorithm: sha256WithRSAEncryption
         30:b6:ae:a0:b4:d5:94:a7:81:33:29:d2:be:ac:40:76:8d:6f:
         d0:f8:9e:9c:77:6f:37:db:a8:85:16:71:80:c8:fe:2c:f9:4e:
         6a:81:be:a0:36:1a:55:72:86:df:fe:e1:f6:e2:36:e2:5c:c0:
         4d:fe:e8:aa:ea:da:65:cb:0c:d3:9f:1a:58:72:58:7f:2b:e5:
         b2:ab:41:63:93:b5:68:84:83:dc:10:b0:0d:f4:f2:c0:ac:fb:
         af:0a:03:da:2f:e3:86:44:a4:c0:e5:79:14:a3:f9:0d:77:48:
         0a:52:4d:a9:e2:af:20:84:fc:1a:33:f3:a4:4d:e7:ee:2f:18:
         cd:1d:4f:fe:d5:55:39:ce:89:a8:9b:93:ed:b3:b1:16:63:26:
         06:da:53:33:2b:8b:f0:a7:ec:af:83:d8:36:67:2b:2b:00:3d:
         42:5d:4a:1e:aa:bc:0c:e0:c0:25:80:36:62:f4:99:e5:93:59:
         43:54:88:d8:a4:48:ba:ee:60:4b:3f:cd:0a:72:7d:9d:94:2e:
         08:d7:bf:c3:40:ce:06:b0:47:a4:78:5a:4a:02:fa:29:30:0c:
         18:11:8b:f7:fd:3f:fc:fa:0e:70:d1:75:13:8e:61:b0:2d:bc:
         f4:d7:51:84:d1:84:2f:34:c5:cf:fa:d3:b4:28:23:a6:ce:ad:
         6c:9f:5a:ee

Comment (UtheMan) [reply parts/linux/cloud-init/artifacts/cse_config.sh:108]: Sure, here: 
```
cat $KUBELET_SERVER_CERT_PATH | openssl x509 -noout -text
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            22:ba:a1:5f:5e:40:95:64:ee:e5:86:8f:88:fe:c7:6e:8d:cc:da:a5
        Signature Algorithm: sha256WithRSAEncryption
        Issuer: CN = aks-nodes-38243387-vmss000000
        Validity
            Not Before: Jan 21 23:38:51 2022 GMT
            Not After : Jan 16 23:38:51 2042 GMT
        Subject: CN = aks-nodes-38243387-vmss000000
        Subject Public Key Info:
            Public Key Algorithm: rsaEncryption
                RSA Public-Key: (2048 bit)
                Modulus:
                    00:b9:ad:af:9f:52:05:fb:6f:a7:bc:6e:cc:ec:0b:
                    86:bc:82:cc:6d:7f:32:67:c3:df:0a:1a:7f:eb:d7:
                    0a:2a:d7:b6:be:20:9f:6c:85:54:bb:82:16:59:d9:
                    30:76:21:3b:3d:b4:6b:98:d1:c0:07:f9:8d:e6:d2:
                    c9:bf:a3:00:90:e7:cc:94:55:49:27:49:14:7c:1d:
                    2c:22:4c:38:a6:92:25:48:82:ec:b4:fb:9c:a0:86:
                    db:0d:6d:cb:62:1d:76:9e:74:93:cb:e9:8c:9a:f1:
                    db:93:1a:d6:ff:b2:c6:3f:56:9c:e4:dc:95:b2:62:
                    9a:08:fa:72:33:ea:f2:ab:ec:fa:c6:a2:df:86:4c:
                    9b:e9:cd:ff:1f:93:2d:83:34:e9:7b:45:95:ff:5d:
                    66:9f:5c:e6:a7:95:f6:c1:53:68:7c:c2:03:61:f1:
                    cf:9a:27:71:09:18:c3:be:4b:b0:ce:ea:a4:4f:ed:
                    53:e1:af:36:81:8f:07:0c:c1:a3:9f:42:71:7c:2c:
                    bb:a5:24:bf:4f:7f:6b:98:46:6e:50:04:9a:f7:cc:
                    8f:b3:9a:f6:7e:8f:cb:0d:b4:b9:40:9b:69:9f:8d:
                    3c:e5:6e:69:bd:04:52:26:b0:32:a7:be:55:1a:a2:
                    af:1b:54:f8:a2:78:65:63:14:7d:b3:94:bb:32:65:
                    04:f1
                Exponent: 65537 (0x10001)
        X509v3 extensions:
            X509v3 Basic Constraints: 
                CA:FALSE
            Netscape Cert Type: 
                SSL Server
            Netscape Comment: 
                OpenSSL Generated Server Certificate
            X509v3 Subject Key Identifier: 
                03:5F:B3:A3:CB:3C:C7:5E:53:92:41:55:E3:71:D4:90:5D:1C:F0:51
            X509v3 Authority Key Identifier: 
                keyid:03:5F:B3:A3:CB:3C:C7:5E:53:92:41:55:E3:71:D4:90:5D:1C:F0:51
                DirName:/CN=aks-nodes-38243387-vmss000000
                serial:22:BA:A1:5F:5E:40:95:64:EE:E5:86:8F:88:FE:C7:6E:8D:CC:DA:A5

            X509v3 Key Usage: critical
                Digital Signature, Key Encipherment
            X509v3 Extended Key Usage: 
                TLS Web Server Authentication
            X509v3 Subject Alternative Name: 
                DNS:aks-nodes-38243387-vmss000000, DNS:10.240.0.4, IP Address:10.240.0.4
    Signature Algorithm: sha256WithRSAEncryption
         30:b6:ae:a0:b4:d5:94:a7:81:33:29:d2:be:ac:40:76:8d:6f:
         d0:f8:9e:9c:77:6f:37:db:a8:85:16:71:80:c8:fe:2c:f9:4e:
         6a:81:be:a0:36:1a:55:72:86:df:fe:e1:f6:e2:36:e2:5c:c0:
         4d:fe:e8:aa:ea:da:65:cb:0c:d3:9f:1a:58:72:58:7f:2b:e5:
         b2:ab:41:63:93:b5:68:84:83:dc:10:b0:0d:f4:f2:c0:ac:fb:
         af:0a:03:da:2f:e3:86:44:a4:c0:e5:79:14:a3:f9:0d:77:48:
         0a:52:4d:a9:e2:af:20:84:fc:1a:33:f3:a4:4d:e7:ee:2f:18:
         cd:1d:4f:fe:d5:55:39:ce:89:a8:9b:93:ed:b3:b1:16:63:26:
         06:da:53:33:2b:8b:f0:a7:ec:af:83:d8:36:67:2b:2b:00:3d:
         42:5d:4a:1e:aa:bc:0c:e0:c0:25:80:36:62:f4:99:e5:93:59:
         43:54:88:d8:a4:48:ba:ee:60:4b:3f:cd:0a:72:7d:9d:94:2e:
         08:d7:bf:c3:40:ce:06:b0:47:a4:78:5a:4a:02:fa:29:30:0c:
         18:11:8b:f7:fd:3f:fc:fa:0e:70:d1:75:13:8e:61:b0:2d:bc:
         f4:d7:51:84:d1:84:2f:34:c5:cf:fa:d3:b4:28:23:a6:ce:ad:
         6c:9f:5a:ee
```

**Review by alexeldeib — APPROVED** [review]
lgtm, thanks!

File: parts/linux/cloud-init/artifacts/cse_config.sh [hunk 2]
```diff
@@ -81,9 +82,37 @@ configureHTTPProxyCA() {
...
+    basicConstraints = CA:FALSE
+    nsCertType = server
+    nsComment = "OpenSSL Generated Server Certificate"
+    subjectKeyIdentifier = hash
+    authorityKeyIdentifier = keyid,issuer:always
+    keyUsage = critical, digitalSignature, keyEncipherment
+    extendedKeyUsage = serverAuth
+    subjectAltName = @alt_names
+    [alt_names]
+    DNS.1 = ${NODE_NAME}
```
Comment (phealy) [inline parts/linux/cloud-init/artifacts/cse_config.sh:107]: Can we please also include the long form hostname somehow?

```shell
root@aks-nodepool1-21278727-vmss000000:/# hostname                                                                                                                                                                   
aks-nodepool1-21278727-vmss000000                                                                                                                                                                                    
root@aks-nodepool1-21278727-vmss000000:/# hostname -f                                                                                                                                                                        
aks-nodepool1-21278727-vmss000000.kqgnwzl40szutf5hh4bauc4byc.xx.internal.cloudapp.net
```

**Pull request merged**